src/main/java/it/gov/pagopa/payhub/auth/utils/JWTValidator.java

```diff
@@ -86,4 +90,27 @@ public void validateInternalToken(String token) {
             throw new InvalidTokenException("The token is not valid");
         }
     }
+
+    /**
+     * Validates JWT signature with publickey.
+     *
+     * @param applicationName the application name for which to validate the token
+     * @param token the JWT to validate
+     * @param publicKey the key to use in the verify instance.
+     * @throws Exception if the token is invalid for any other reason
+     *         (e.g., signature verification failure).
+     */
+    public Pair<String, Map<String, Claim>> validateLegacyToken(String applicationName, String token, PublicKey publicKey) {
+        try{
+            DecodedJWT jwt = JWT.decode(token);
+
+            Algorithm algorithm = Algorithm.RSA512((RSAPublicKey) publicKey, null);
```

Comment:
are we sure that the algorithm will be always RSA512?
If not, we could read the algorithm from the jwt header, using RSA512 as default one if not recognized from header

Reply:
Yes, take a look at it.regioneveneto.mygov.payment.mypay4.security.JwtTokenUtil#main